Fix failing Traeffik tests...

davidfestal opened the pull request:
Signed-off-by: David Festal <dfestal@redhat.com>

davidfestal committed:
Fix failing Traeffik tests...

... by:
- adding the new `CHE_IS_DEV_MACHINE` env variable in tests
- pulling up all the `CustomServerEvaluationStrategy` features in an
abstract `BaseServerEvaluationStrategy` (which all other Che strategies
extend) and have the `CustomServerEvaluationStrategy` class simply
extend this `BaseServerEvaluationStrategy`.

Signed-off-by: David Festal <dfestal@redhat.com>

benoitf commented on plugins/plugin-traefik/plugin-traefik-docker/src/test/java/org/eclipse/che/plugin/traefik/TraefikCreateContainerInterceptorTest.java:
```diff
@@ -115,8 +115,7 @@ protected void setup() throws Exception {
         when(containerConfig.getExposedPorts()).thenReturn(containerExposedPorts);
         when(imageInfoConfig.getExposedPorts()).thenReturn(imageExposedPorts);
 
-
```
maybe remove this changed line so test class is not updated ;-)

davidfestal replied:
sure !

davidfestal replied:
I made the change, even if the class is still changed in another place

davidfestal committed:
Fix tests in the LocalDockerEvaluationStrategy...

... by correctly using the boolean attribute to manage the new use-case
introduced by @fbenoit in master.

Signed-off-by: David Festal <dfestal@redhat.com>
